pilot/pkg/model/service.go

```diff
@@ -159,6 +159,12 @@ const (
 	k8sSeparator = "."
 )
 
+const (
+	TunnelLabel          = "networking.istio.io/tunnel"
```

Comment:
I think we need a domain for k8s best practices. Maybe we should buy hbone.io...

Also, somewhat related. I was thinking instead of `tunnel=h2`... we should have it be future proof

In the future we may want http/1.1 support; we probably don't need to say `tunnel=h2,h1` since we can negotiate over ALPN. So no need to change the label there.. but `h2` is also probably a bad name.

But if we add HTTP/3, we cannot use ALPN negotiation. We can use alt-svc but seems wasteful when we could know ahead of time (and also ~broken to ever use http/2 for tunneling UDP), we likely we need a way to say `tunnel=h2,h3`.

So basically:
* Label should support multiple CSV values (like ALPN somewhat)
* `h2` is probably a bad name, but I am not sure a better one.
* We can switch the base of the label but it ought to be a domain we own.
WDYT?

Reply:
Agree with @costinm . This needs to be a declaration that a TCP port is open for mTLS and let ALPN decide the codec. We'll need a separate declaration for QUIC port. And another declaration that regular ports are closed.

Reply:
Not sure about the last one - it's not our concern if NetworkPolicy or some other setting is closing regular ports. 

What we want to know is:
- if a pod is using ztunnel 
- if a pod is using sidecar with ztunnel support

Forget h2 - once TLS handshake happen we can negotiate what protocol we use. 

Since ztunnel should not implement the old protocol - maybe it doesn't even matter to indicate a sidecar with 
ztunnel support. By the time we roll ztunnel, we can assume 1-2 versions of istio are done and all 
sidecars accept ztunnel.